### PR TITLE
Add a root cause arg to the fallback converter

### PR DESCRIPTION
**Pull request by bhack**
Please check https://github.com/keras-team/keras-cv/pull/146#issuecomment-1063974863

Any improvement feedback on the `root_cause` message description and fix hints to the developer is appreciated.

Side note: I am going to "abuse" the CI here a little bit as I'am not compiling and testing this locally cause the build overhead on every PR with the daily [LLVM updates](https://discuss.tensorflow.org/t/llvm-updates-and-bazel-cache/2060) and cause [I cannot use the nightly remote cache](https://github.com/tensorflow/build/issues/72)

### CODE REVIEW

File: tensorflow/python/ops/parallel_for/pfor.py [hunk 1]
```diff
@@ -4659,7 +4667,7 @@ def _init_values(self):
           # vectorization" format, so we want to keep it that way as much as
           # possible. We'll accumulate finished iterations (only relevant for
           # pfor-loop-variant while_loop conditions) in an accumulator with
-          # type:
+          # type :
```
Comment (wangpengmit) [inline tensorflow/python/ops/parallel_for/pfor.py:4670]: nit: unnecessary change.

File: tensorflow/python/ops/parallel_for/pfor.py [hunk 2]
```diff
@@ -1612,14 +1613,20 @@ def _add_to_stack(x):
           pfor_inputs = _PforInput(self, y_op, converted_inputs)
           try:
             try:
-              new_outputs = converter(pfor_inputs)
+              if (converter.__name__ == "_fallback_converter"):
```
Comment (wangpengmit) [inline tensorflow/python/ops/parallel_for/pfor.py:1616]: A better approach is to do `converter = functools.partial(_fallback_converter, root_cause="...")` in line https://github.com/tensorflow/tensorflow/blob/4c464afd3d48e175d32f16a2d67c4902ba6624fb/tensorflow/python/ops/parallel_for/pfor.py#L1601

And the root cause should be `"has no variant outputs and has no vectorized variant inputs"`.

File: tensorflow/python/ops/parallel_for/pfor.py [hunk 3]
```diff
@@ -1612,14 +1613,20 @@ def _add_to_stack(x):
...
+              else: 
+                new_outputs = converter(pfor_inputs)
             except ConversionNotImplementedError as e:
               has_vectorized_variant_inputs = any(
                   _is_variant_with_internal_stacking(x) for x in
                   y_op.inputs)
               if (self._fallback_to_while_loop
                   and not has_vectorized_variant_inputs):
-                new_outputs = _fallback_converter(pfor_inputs)
+                root_cause = "has not a vectorized variant inputs"
```
Comment (wangpengmit) [inline tensorflow/python/ops/parallel_for/pfor.py:1627]: nit: `"has no vectorized variant inputs"`.

File: tensorflow/python/ops/parallel_for/pfor.py (hunk 1 above)
Comment (bhack) [reply tensorflow/python/ops/parallel_for/pfor.py:4670]: It is required for https://github.com/python/typed_ast/issues/62. I don't know if it was released in a more recent `pylint`release.

**Comment by bhack** [comment]
Can we in general give a better self-explainable string to avg developer?
I think that the currents strings are a little bit cryptic and we don't have any hints on what the user need to check/change to workaround the fallback.

File: tensorflow/python/ops/parallel_for/pfor.py (hunk 1 above)
Comment (bhack) [reply tensorflow/python/ops/parallel_for/pfor.py:4669]: I've checked and it is not fixed in any release:
https://github.com/PyCQA/pylint/issues/4028#issuecomment-761575189

File: tensorflow/python/ops/parallel_for/pfor.py [hunk 4]
```diff
@@ -1613,18 +1616,14 @@ def _add_to_stack(x):
...
-                new_outputs = converter(pfor_inputs)
+              new_outputs = converter(pfor_inputs)
             except ConversionNotImplementedError as e:
               has_vectorized_variant_inputs = any(
                   _is_variant_with_internal_stacking(x) for x in
                   y_op.inputs)
               if (self._fallback_to_while_loop
                   and not has_vectorized_variant_inputs):
-                root_cause = "has not a vectorized variant inputs"
+                root_cause = "has no a vectorized variant inputs"
```
Comment (wangpengmit) [inline tensorflow/python/ops/parallel_for/pfor.py:1626]: Remove "a" in "has no a".

**Review by wangpengmit — APPROVED** [review]

**Comment by wangpengmit** [comment]
> Can we give a better self-explainable string to the "avg" developer? I think that the current strings are a little bit cryptic and we don't have any hints on what the user need to check/change to workaround the fallback.

Yeah good point. Even I don't understand the strings. I don't have a quick suggestion. Please file a bug against them so we can look into pfor's UX issue closer.

File: tensorflow/python/ops/parallel_for/pfor.py (hunk 4 above)
Comment (bhack) [reply tensorflow/python/ops/parallel_for/pfor.py]: Done

**Comment by bhack** [comment]
> > Can we give a better self-explainable string to the "avg" developer? I think that the current strings are a little bit cryptic and we don't have any hints on what the user need to check/change to workaround the fallback.
> 
> Yeah good point. Even I don't understand the strings. I don't have a quick suggestion. Please file a bug against them so we can look into pfor's UX issue closer.

Done: https://github.com/tensorflow/tensorflow/issues/55425

**Review by wangpengmit — APPROVED** [review]

**Comment by bhack** [comment]
@wangpengmit Can you check why feedback/copybara is failing?
